mk/travis.sh

```diff
@@ -64,17 +64,31 @@ rustc --version
 
 if [[ "$MODE_X" == "RELWITHDEBINFO" ]]; then mode=--release; fi
 
-CC=$CC_X CXX=$CXX_X cargo build -j2 ${mode-} --verbose --target=$TARGET_X
+if [[ "$KCOV" == "1" ]]; then
+  norun=--no-run
+  RUSTFLAGS_X="-C link-dead-code"
+  ./mk/travis-install-kcov.sh
+fi
+
+CC=$CC_X CXX=$CXX_X RUSTFLAGS=${RUSTFLAGS_X-} cargo build -j2 ${mode-} --verbose --target=$TARGET_X
 
 case $TARGET_X in
 arm-unknown-linux-gnueabi|aarch64-unknown-linux-gnu)
   ;;
 *)
-  CC=$CC_X CXX=$CXX_X cargo test -j2 ${mode-} --verbose --target=$TARGET_X
-  CC=$CC_X CXX=$CXX_X cargo doc -j2 ${mode-} --verbose --target=$TARGET_X
+  CC=$CC_X CXX=$CXX_X RUSTFLAGS=${RUSTFLAGS_X-} cargo test -j2 ${mode-} ${norun-} --verbose --target=$TARGET_X
```

Comment:
I think it would be better to run the tests with `cargo test` like normal, and then run them again under `kcov`. That way, we measure whether the tests pass/fail without kcov having any effect on them, which I think is important.